### PR TITLE
Control tests

### PR DESCRIPTION
**Pull request by BrianBreniser**
I added some tests, and did some minor code cleanup, although most of them won't work if there isn't a real gpio device, so we will have to test them on the LED prototype. The ones that do pass ensure that a few simple things are working properly every time tests are run.

Feedback is appreciated. I'll merge these changes if someone checks them off.

### CODE REVIEW

**Comment by BrianBreniser** [comment]
I'm confused as to why travis is failing, it builds and tests fine on my box, anyone else having build issues? Or know more about travis than I to know what is going on?

**Comment by BrianBreniser** [comment]
broken test found and removed, thanks Harrison for catching that.

**Comment by cwalexand** [comment]
Both modes introduce new warnings into the build. Below are the results for test_mode. The same warnings occur for flight_mode but have different line numbers.

src/gpio.rs:147:9: 147:14 warning: variable does not need to be mutable, #[warn(unused_mut)] on by default
src/gpio.rs:147     let mut p = MyPins::new();

src/gpio.rs:165:9: 165:10 warning: unused variable: `v`, #[warn(unused_variables)] on by default
src/gpio.rs:165     let v = p.get_value(42).unwrap();

src/gpio.rs:173:5: 173:24 warning: unused result which must be used, #[warn(unused_must_use)] on by default
src/gpio.rs:173     p.set_value(42, 0);

---

Both modes ignore the following tests:

test gpio::test_multiple_pins ... ignored
test gpio::test_set_pin ... ignored
test timestep ... ignored

**Comment by BrianBreniser** [comment]
I'll check out the new warnings and it will likely be quick fixes. The ignored tests are tests that won't pass if not on the edison board, so I ignored them using #[ignore] and they can be run by running cargo test --ignored. Usually ignore is for long running tests but they can be for anything that doesn't have to run every time. I don't know of another way to add a separate test flags(such as cargo run --edison would be nice), so any suggestions would be awesome.

**Comment by BrianBreniser** [comment]
Tuesday I'll make more progress on this, and fix the merge conflicts.

**Comment by BrianBreniser** [comment]
rebasing was annoying but there was a git merge that caused all kinds of problems. I think I fixed everything and the tests finally pass on both flight and test mode. wrapper.rs has nearly all the warnings, and they are all dead code warnings, but I'm letting Sohail poke at that before making any changes. I think these are ready for integration to master now.

File: src/common/control.rs [hunk 1]
```diff
@@ -28,147 +25,135 @@ pub struct Control {
...
-        try!(self.write_pin(CCW, value, mem));
-    } // ELSE
-    else {
-        // turn off both gpio pins and update their state
-        // CALL write_pin(CW_pin, 0, mem)
-        // CALL write_pin(CCW_pin, 0, mem)
-        try!(self.write_pin(CW, 0, mem));
-        try!(self.write_pin(CCW, 0, mem));
-    } // END IF
-
```
Comment (chrisliebert) [inline src/common/control.rs:98]: by replacing // with /// you can take advantage documentation building with 'cargo doc'

Comment (BrianBreniser) [inline src/common/control.rs:98]: Were you referring to comments on the functions themselves? Or somewhere in the block? Currently common has a different comment type, but it's consistent right now so I didn't touch it. I was thinking about changing to the /// type.

Comment (cwalexand) [inline src/common/control.rs:98]: I think that rustdoc comments work on libraries but not binaries. So, the common files can have // style comments, but test_mode/src and flight_mode/src should use /// style comments.

**Pull request merged**